sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/ManagedServiceIdentityType.cs

```diff
@@ -17,17 +17,12 @@ internal partial class ManagedServiceIdentityTypeConverter : JsonConverter<Manag
         {
             public override void Write(Utf8JsonWriter writer, ManagedServiceIdentityType model, JsonSerializerOptions options)
             {
-                writer.WritePropertyName("type");
                 writer.WriteStringValue(model.ToString());
             }
             public override ManagedServiceIdentityType Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
             {
                 using var document = JsonDocument.ParseValue(ref reader);
-                foreach (var property in document.RootElement.EnumerateObject())
-                {
-                    return new ManagedServiceIdentityType(property.Value.GetString());
-                }
-                return null;
+                return new ManagedServiceIdentityType(document.RootElement.GetString());
```

Comment:
This whole "model" is just a string, it should serialize into a string, and deserialize from a string.
Therefore here the `RootElement` is just the string value